streams/src/main/java/org/apache/kafka/streams/processor/TopologyDescription.java

```diff
@@ -0,0 +1,476 @@
+
+        Subtopology(final int id,
+                    final Set<Node> nodes) {
+            this.id = id;
+            this.nodes = nodes;
+        }
+
+        /**
+         * Internally assigned unique ID.
+         * @return the ID of the sub-topology
+         */
+        public int id() {
+            return id;
+        }
+
+        /**
+         * All nodes of this sub-topology.
+         * @return set of all nodes within the sub-topology
+         */
+        public Set<Node> nodes() {
+            return Collections.unmodifiableSet(nodes);
+        }
+
+        @Override
+        public String toString() {
+            return "Sub-topology: " + id + "\n" + nodesAsString();
+        }
+
+        private String nodesAsString() {
+            final StringBuilder sb = new StringBuilder();
+            for (final Node node : nodes) {
+                sb.append("    ");
+                sb.append(node);
+                sb.append('\n');
+            }
+            return sb.toString();
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            final Subtopology that = (Subtopology) o;
+            return id == that.id
+                && nodes.equals(that.nodes);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(id, nodes);
+        }
+    }
+
+    /**
+     * Represents a {@link Topology#addGlobalStore(StateStoreSupplier, String,
+     * org.apache.kafka.common.serialization.Deserializer, org.apache.kafka.common.serialization.Deserializer, String,
+     * String, ProcessorSupplier)} global store}.
+     * Adding a global store results in adding a source node and one stateful processor node.
+     * Note, that all added global stores form a single unit (similar to a {@link Subtopology}) even if different
+     * global stores are not connected to each other.
+     * Furthermore, global stores are available to all processors without connecting them explicitly, and thus global
+     * stores will never be part of any {@link Subtopology}.
+     */
+    public final static class GlobalStore {
+        private final Source source;
+        private final Processor processor;
+
+        GlobalStore(final String sourceName,
+                    final String processorName,
+                    final String storeName,
+                    final String topicName) {
+            source = new Source(sourceName, topicName);
+            processor = new Processor(processorName, Collections.singleton(storeName));
+            source.successors.add(processor);
+            processor.predecessors.add(source);
+        }
+
+        /**
+         * The source node reading from a "global" topic.
+         * @return the "global" source node
+         */
+        public Source source() {
```

Comment:
I wonder if we should return an interface here, i.e., once that only has the read/get methods? `Source` has `addPredecessor` and `addSuccessor`, which i don't think is correct for the user facing interfaces.
Same for the next method.